File: Labs/Lab 1/webUI-test/src/test/java/com/accessHQ/models/Planet.java
package com.accessHQ.models;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Planet {
    private final WebElement planetWebElement;

    public Planet(WebElement planetWebElement) {
        this.planetWebElement = planetWebElement;
    }

    public void clickExploreButton() {
        planetWebElement.findElement(By.tagName("button")).click();
    }

    public String getName() {
        return planetWebElement.findElement(By.className("name")).getText();
    }
}

File: Labs/Lab 2/AccessHQ.NETTestProject/Models/Planet.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AccessHQ.NETTestProject.Models
{
    internal class Planet
    {
        private WebElement planetWebElement;

        public Planet(WebElement planetWebElement)
        {
            this.planetWebElement = planetWebElement;
        }

        public String getName()
        {
            return planetWebElement.FindElement(By.ClassName("name")).Text.ToLower();
        }

        public void ClickExplore()
        {
            planetWebElement.FindElement(By.TagName("button")).Click();
        }
    }
}
